tidy3d/material_library/parametric_materials.py

```diff
@@ -0,0 +1,266 @@
+
+
+class Graphene(ParametricVariantItem2D):
+    """Surface conductivity model for graphene, including intraband and interband terms,
+    as described in
+
+    George W. Hanson, "Dyadic Green's Functions for an Anisotropic,
+    Non-Local Model of Biased Graphene," IEEE Trans. Antennas Propag.
+    56, 3, 747-757 (2008).
+
+    Example
+    -------
+    >>> medium = Graphene(mu_c = 0.2).medium
+
+    """
+
+    gamma: float = pd.Field(
+        GAMMA_DEF,
+        title="Scattering rate in eV",
+        description="Scattering rate in eV.",
+    )
+    mu_c: float = pd.Field(
+        MU_C_DEF,
+        title="Chemical potential in eV",
+        description="Chemical potential in eV.",
+    )
+    temp: float = pd.Field(
+        TEMP_DEF,
+        title="Temperature in K",
+        description="Temperature in K.",
+    )
+    include_interband: bool = pd.Field(
+        True,
+        title="Include interband terms",
+        description="Include interband terms, relevant at high frequency (IR). "
+        "Otherwise, the intraband terms only give a simpler Drude-type model relevant "
+        "only at low frequency (THz).",
+    )
+
+    @property
+    def medium(self) -> Medium2D:
+        """Surface conductivity model for graphene."""
+        intraband = self.drude_intraband
+        if self.include_interband:
+            interband = self.pole_residue_interband
+            intraband_poles = intraband.pole_residue.poles
+            interband_poles = interband.pole_residue.poles
+            poles = intraband_poles + interband_poles
+            pole_residue = PoleResidue(poles=poles)
+            return Medium2D(ss=pole_residue, tt=pole_residue)
+        return Medium2D(ss=intraband, tt=intraband)
+
+    @property
+    def drude_intraband(self) -> Drude:
+        """A Drude-type model for the intraband term of graphene.
+
+        Returns
+        -------
+        :class:`Drude`
+            A Drude-type model for the intraband term of graphene.
+
+        """
+        intra_f1 = np.sqrt(
+            (Q_e * K_B * self.temp / (HBAR**2 * 4 * np.pi**3 * EPSILON_0))
+            * (
+                self.mu_c / (K_B * self.temp)
+                + 2 * np.log(np.exp(-self.mu_c / (K_B * self.temp)) + 1)
+            )
+        )
+        intra_delta1 = (1 / HBAR) * self.gamma / np.pi
+        return Drude(coeffs=[(intra_f1, intra_delta1)])
+
+    @property
+    def pole_residue_interband(self) -> PoleResidue:
+        """A pole-residue model for the interband term of graphene. Note that this does not
+        include the intraband term, which is added in separately.
+
+        Returns
+        -------
+        :class:`.PoleResidue`
+            A pole-residue model for the interband term of graphene.
+        """
+
+        mu_c_freq = self.mu_c / (np.pi * HBAR)
+        temp_freq = K_B * self.temp / (np.pi * HBAR)
+
+        freq1 = mu_c_freq
+        freq2 = mu_c_freq + 5 * temp_freq
+
+        ind1 = int(
```

Comment:
probably can refactor this a bit for readability and to avoid duplication.
```py
min_freq_hz = MIN_FREQ_THZ * 1e12
max_freq_hz = MAX_FREQ_THZ * 1e12
ind1, ind2 = [NUM_FREQS * (freq - min_freq_hz) / (max_freq_hz - min_freq_hz) for freq in (freq1, freq2)]

```